agent/agent-tooling/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/logging/DiagnosticTelemetryPipelineListener.java

```diff
@@ -44,12 +44,13 @@ public class DiagnosticTelemetryPipelineListener implements TelemetryPipelineLis
   private static final AtomicBoolean friendlyExceptionThrown = new AtomicBoolean();
 
   private final OperationLogger operationLogger;
-  private final boolean shouldLogWarnings;
+  private final boolean suppressWarningsOnRetryableFailures;
```

Comment:
are you going to add another boolean to suppress logging for other error types? I think error type is irrelevant in this case.

Reply:
the idea here is to suppress retryable failures, since those will be retried later anyways (and then we will log if the retry fails)